cylc/flow/tui/overlay.py

```diff
@@ -473,6 +478,67 @@ def open_log(*_, filename=None, close=False):
             if close:
                 app.close_topmost()
 
+    def open_in_editor(*_, command):
+        """Suspend Tui, open the file in an external utility, then restore Tui.
+
+        Args:
+            command:
+                The command to run as a list, e.g. 'gvim -f'.
+                This command must be blocking, the tui session will be
+                restored when the command exits.
+
+        """
+        import shlex
+        from subprocess import Popen
+        import stat
+        import tempfile
+
+        nonlocal _filename
+
+        try:
+            host, path, text = get_log(_filename)
+        except Exception as exc:
+            host_widget.set_text(f'Error: {exc}')
+            file_widget.set_text('')
+        else:
+            host_widget.set_text(f'Host: {host}')
+            file_widget.set_text(f'Path: {path}')
+
+        with tempfile.NamedTemporaryFile('w+') as temp_file:
+            # write the text into a temp file
+            temp_file.write(text)
+            temp_file.seek(0, 0)
+
+            # make the file readonly to avoid confusion
+            os.chmod(temp_file.name, stat.S_IRUSR)
```

Comment:
This is the approach we used to use `cylc view`. Dump the text into a temp file and mark it as readonly so that people don't think they can edit it.

Note, we can't reliably edit the file locally because it might not be a local file.